### DISCUSSION TITLE
how to write the class hooks name for the model properties ?

### DISCUSSION THREAD

**Discussion by jichiduo** [livewire/livewire#4746] [1. Help] [answered]
Hi ,

As the document https://laravel-livewire.com/docs/2.x/lifecycle-hooks showed , I can write some logic when one single property updated by using class hooks , like 
``` php
class HelloWorld extends Component
{
    public $foo;
 
    public function updatedFoo($value)
    {
        //logic here
       $this->foo = "Hi ".$value;
    }
 
 } 
```
it works well in my code , now I want to use the same method to modify one property of  a model , how can I write the name for the updated hook or is there a method can do this ? many thanks for your kindly advice
``` php
class Post extends Model
{
    protected $table="posts";
    protected $fillable = ['title', 'content'];
}

class HelloWorld extends Component
{
    public $post = new Post();
 
    public function updatedPost_Title($value)
    {
        //this wont work
        $this->post->title = "Hi ". $value;
    }
 
 } 
```

**Comment by zepfietje** [comment, marked as answer]
Hey @jichiduo!

I think it should work without underscore, like this:
```php
public function updatedPostTitle($value)
{
}
```

**Reply by joshhanley** [reply]
@jichiduo yeah @zepfietje is correct, try it without the underscore.

If you have a look on that [lifecycle-hooks](https://laravel-livewire.com/docs/2.x/lifecycle-hooks) page you linked, look at the `updatedFooBar` hook example.

Hope this helps!

**Comment by jichiduo** [comment]
Hi @zepfietje , @joshhanley 

It works well , thanks for the kindly help
Have a good day, stay safe

**Reply by joshhanley** [reply]
No worries! 🙂
